nlp/optimize/tron.py

```diff
@@ -362,7 +370,7 @@ def projected_linesearch(self, x, l, u, g, d, H, alpha=1.0):
 
         # Compute the final iterate and step.
         s = projected_step(x, alpha * d, l, u)
-        x = project(x + alpha * s, l, u)
+        x = project(x + s, l, u)
```

Comment:
Read the code carefully. The original version said 
```python 
x = project(x + alpha * s, l, u)
```
while your writing above says
```python 
x = project(x + alpha * d, l, u)
```
(s replaced by d) so there was definitely a bug. My use of the project() function in
```python
x = project(x + s, l, u)
```
was just to prevent an epsilon-violation of the variable bounds. I could have just said 
```python
x = x + s 
```
for that line. If you prefer `x + alpha * d` over `x + s` in the project call, that's easy enough to fix.